anoncreds/src/data_types/anoncreds/rev_reg.rs

```diff
@@ -47,3 +52,130 @@ impl Validatable for RevocationRegistryDelta {}
 pub struct RevocationRegistryDeltaV1 {
     pub value: ursa::cl::RevocationRegistryDelta,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RevocationList {
+    rev_reg_id: RevocationRegistryId,
+    #[serde(with = "serde_revocation_list")]
+    revocation_list: bitvec::vec::BitVec,
+    #[serde(flatten)]
+    registry: ursa::cl::RevocationRegistry,
+    timestamp: u64,
+}
+
+impl From<&RevocationList> for ursa::cl::RevocationRegistry {
+    fn from(rev_reg_list: &RevocationList) -> ursa::cl::RevocationRegistry {
+        rev_reg_list.registry.clone()
+    }
+}
+
+impl RevocationList {
+    pub(crate) fn timestamp(&self) -> u64 {
+        self.timestamp
+    }
+
+    pub(crate) fn state(&self) -> &bitvec::vec::BitVec {
+        &self.revocation_list
+    }
+
+    pub(crate) fn state_owned(&self) -> bitvec::vec::BitVec {
+        self.revocation_list.clone()
+    }
+
+    pub fn new(
+        rev_reg_id: &str,
+        revocation_list: bitvec::vec::BitVec,
+        registry: ursa::cl::RevocationRegistry,
+        timestamp: u64,
+    ) -> Result<Self, error::Error> {
+        Ok(RevocationList {
+            rev_reg_id: RevocationRegistryId::new(rev_reg_id)?,
+            revocation_list,
+            registry,
+            timestamp,
+        })
+    }
+}
+
+pub mod serde_revocation_list {
+    use super::*;
+    pub fn serialize<S>(state: &bitvec::vec::BitVec, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut seq = s.serialize_seq(Some(state.len()))?;
+        for element in state {
+            let e = match *element.as_ref() {
+                true => 1,
+                false => 0,
+            };
+            seq.serialize_element(&e)?;
+        }
+        seq.end()
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<bitvec::vec::BitVec, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct JsonBitStringVisitor;
+
+        impl<'de> Visitor<'de> for JsonBitStringVisitor {
+            type Value = bitvec::vec::BitVec;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a seq containing revoation state, i.e. [1, 0, 1]")
+            }
+
+            fn visit_seq<S>(self, mut v: S) -> Result<Self::Value, S::Error>
+            where
+                S: SeqAccess<'de>,
+            {
+                // TODO: do we have a min size for this?
+                let mut bv = BitVec::with_capacity(v.size_hint().unwrap_or_default());
```

Comment:
This is the default for `usize`, so it depends on compiling target. 
But it will be better if we know  some min size

Reply:
Wait so it would be USIZE::MAX? That's quite big :) Maybe we can just initialise it with a dynamic size? Not sure if BitVec supports that but I assume so.

Reply:
hm. no actually it is 256 bits now.

This is AFAIK just the initial allocation, it will be reallocate as it gets filled, like Vec. 
If we know the capacity in advance then it just avoids reallocating.

Reply:
Ah yeah okay.